src/robot/libraries/BuiltIn.py

```diff
@@ -2526,6 +2527,64 @@ def _get_suite_in_teardown(self, kwname):
                                % kwname)
         return self._context.suite
 
+    @run_keyword_variant(resolve=0, dry_run=True)
+    async def gather_async_keywords(self, *keywords):
```

Comment:
I'm not sure is this keyword a good addition to the BuiltIn library. It looks like something that may need to evolve in the future, and making such changes in BuiltIn is hard. It also looks like there would be needs for more similar keywords and having all async keywords in a single library would be better.

Instead of having this keyword in BuiltIn, I believe it would be better to have a separate Async library. In the beginning it would probably be best if it would be an external library, because that allows faster development independent from Robot's release schedule. External libraries don't have such strict backwards compatibility concerns as Robot's standard libraries either. Once the library gets more mature and proves generally useful, it could then be included as one of the standard libraries.

Reply:
Makes sense to be a library. I can create even more keywords there to cover asyncio functionality, will see to it soon.